chromadb/test/conftest.py

```diff
@@ -982,3 +995,77 @@ def log_tests(request: pytest.FixtureRequest) -> Generator[None, None, None]:
     yield
 
     logger.debug(f"Finished test: {test_name}")
+
+
+@pytest.fixture(scope="session")
+def embedding_function_dependencies() -> Generator[None, None, None]:
+    """
+    Fixture to install and uninstall dependencies required for embedding function tests.
+    This fixture has session scope to ensure dependencies are installed only once
+    for the entire test session and cleaned up afterward.
+    """
+    # List of packages to install
+    packages = [
+        "openai",
+        "cohere",
+        "sentence_transformers",
+        "google-generativeai",
+        "ollama",
+        "pillow",
+        "voyageai",
+        "open-clip-torch",
+        "text2vec",
+        "InstructorEmbedding",
+    ]
+
+    # Install packages
+    logger.info("Installing embedding function dependencies...")
+    for package in packages:
+        try:
+            subprocess.check_call([sys.executable, "-m", "pip", "install", package])
```

Comment:
nit: not sure how slow this is but might be faster if you give all packages to `pip install` at once